Added doc: host setup - disable lvmetad (#296)
piraeusdatastore/piraeus-operator: doc/host-setup.md

CentOS 7 and Ubuntu 18.04 by default use `lvmetad`,  a metadata caching daemon for LVM. The daemon improves performance of LVM commands by avoiding rescanning the disks on every execution. `piraeus-ns-node` pods run LVM commands inside a container without access to lvmetad. Since that means  `lvmetad` would get out of sync, it is recommended to disable `lvmetad` in CentOS 7 and Ubuntu 18.04 Bionic.

Note: Newer distributions no longer include with `lvmetad`, no changes necessary.

Follow below steps to disable lvmetad completely:
```bash
$ systemctl stop lvm2-lvmetad.socket lvm2-lvmetad.service

$ systemctl disable lvm2-lvmetad.service

$ systemctl mask lvm2-lvmetad.socket

$ sed -i 's/use_lvmetad = 1/use_lvmetad = 0/' /etc/lvm/lvm.conf

$ cat /etc/lvm/lvm.conf | grep use_lvmetad
use_lvmetad = 0
```
If the root partition already uses LVM, you also need to update the initial ram-disk:
```bash
# CentOS 7
$ cp -vf /boot/initramfs-$(uname -r).img /boot/initramfs-$(uname -r).img.$(date +%m-%d-%H%M%S).bak
$ dracut -f -v

# Ubuntu 18.04
$ cp -vf /boot/initrd.img-$(uname -r) /boot/initd.img.$(uname -r).$(date +%m-%d-%H%M%S).bak
$ update-initramfs -c -k $(uname -r)
$ update-grub
```

> 

Reviewer: Weird that this wasn't caught by your `pre-commit` run, but this still fails here. Just remove this line, I think thats the last blocker

Author: As per vscode markdownlint suggestion, I added blank lines before and after code block. `git commit` passed with no error per `pre-commit`: 

```
 git commit -s -m 'clear syntax errors posted by markdownlint'
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...........................................(no files to check)Skipped
check for added large files..............................................Passed
gofumpt..............................................(no files to check)Skipped
golangci-lint........................................(no files to check)Skipped
go-unit-tests........................................(no files to check)Skipped
generated deep-copy code must be up-to-date..........(no files to check)Skipped
generated crds must be up-to-date....................(no files to check)Skipped
generate helm values for CN users....................(no files to check)Skipped
generate full yaml deployment from helm templates....(no files to check)Skipped
[master be83e4e] clear syntax errors posted by markdownlint
 1 file changed, 4 insertions(+), 2 deletions(-)
```
Please see if it can be merged.

# cat /sys/module/drbd/parameters/usermode_helper
disabled
```

> ## Disable Lvmetad in CentOS 7 and Ubuntu 18.04

Reviewer: ```suggestion
## Disable Lvmetad on CentOS 7 and Ubuntu 18.04
```

> Reference: https://access.redhat.com/solutions/2053483

Reviewer: This link requires logging in with a red hat account. I think there is enough information so we can remove it.

CentOS 7 and Ubuntu 18.04 by default use **lvmetad**,  a metadata caching daemon for LVM. The daemon receives notifications from udev rules. Through these notifications, lvmetad has an up-to-date and consistent image of the volume groups available in the system. lvmetad improves performance of LVM commands by saving them from scanning the disks in every execution. 

> `piraeus-ns-node` pod runs lvm commands inside a container without using any lvmetad or udev. And it does not communicate to lvmetad daemon in the operation system. Therefore, users must run `pvscan --cache` in the OS each time to view LVM changes after a Piraeus volume operation. 

Reviewer 2: Haven't we considering passing through `/etc/lvm.conf` and `/run/lvm` from host into container to avoid such problems?

Author: Yes, I have another incoming PR to support lvmetad just for centos 7 and ubuntu bionic as we also hear from some conservative users who do not want to turn lvmetad off. But so far, disabling lvmetad is the simplest solution and is consistent with the lvm trend in centos 8 and ubuntu focal.

Author: Passing `/etc/lvm.conf` and `/run/lvm` is not adequate to enable lvm binary communicate to os-level lvmetad daemon from within `piraeus-ns-node` container.  Only lvm binary lvm2-2.02 binary in centos/ubi 7, ubuntu bionic, and debian strech base images still have support for lvmetad ( they still having `use_lvmetad = 1` in lvm.conf).  `piraeus-sever` is currently based on debian bullseye (presviously buster) image which uses lvm2-2.03 that has deprecated lvmetad support. Therefore, to support lvmetad, `piraeus-server` needs to have a version based on debian stretch (or ubuntu bionic or centos 7) to be able to install lvm2-2.02.  Apart from that, /run/udev is also needed for lvmetad.

Author: Unfornately, linbit does not maintain a repo for debian stretch any more. Therefore, I managed to rebase `piraeus-server`to ubuntu bionic image, and successfully had lvm cmd updating lvmetad in os.

In CentOS 8 and in Ubuntu 20.04 (focal), lvmetad is deprecated because lvm2 has removed it since version 2.03. **Therefore, it is also recommended to disable lvmetad in CentOS 7 and Ubuntu 18.04 binoic**. 

> Follow below steps to disable lvmetad completely: 

Reviewer: ```suggestion
Follow below steps to disable lvmetad completely:
```

There is a trailing whitespace here

```bash
$ systemctl stop lvm2-lvmetad.socket lvm2-lvmetad.service

$ systemctl disable lvm2-lvmetad.service

> $ systemctl mask lvm2-lvmetad.socket

Reviewer 2: Why to mask without disabling lvm2-lvmetad.socket?

Author: It is also recommend by RedHat's article "[https://access.redhat.com/solutions/2053483] https://access.redhat.com/solutions/2053483"

$ sed -i 's/use_lvmetad = 1/use_lvmetad = 0/' /etc/lvm/lvm.conf

$ cat /etc/lvm/lvm.conf | grep use_lvmetad
use_lvmetad = 0
```

> If lvm is used for root, initial RAM has to be updated:

Reviewer: ```suggestion
If the root partition already uses LVM, you also need to update the initial ram-disk:
```

> If lvm is used for root, initial RAM has to be updated:

Reviewer 2: Is it really necessary step? You already disabled `lvm2-lvmetad.service` and `lvm2-lvmetad.socket`, and configured `lvm.conf` to not use it.

Why do we need any changes on initramfs phase?

Author: It is recommended by Redhat to load lvm.conf properly during boot, specifically for root on lvm. Apart from that, in CentOS 7 without updated intramfs, commends such as vgdisplay will still complain about missing lvmetad.
